### PR TITLE
Add wasm support to jaeger

### PR DESCRIPTION
**Pull request by boxdot**
This PR add another feature `wasm_collector_client` to the `opentelemetry-jaeger` crate which enable a WebAssembly collector.

Note: At the moment of writing Jaeger does not support Option Method requests, that are required by CORS: https://developer.mozilla.org/en-US/docs/Glossary/Preflight_request. A simple solution is to put Jaeger behind a proxy and handle the request in the proxy.

### CODE REVIEW

File: opentelemetry-jaeger/src/collector.rs [hunk 1]
```diff
@@ -1,88 +1,247 @@
...
 
 /// `CollectorAsyncClientHttp` implements an async version of the
 /// `TCollectorSyncClient` interface over HTTP
 #[derive(Debug)]
 pub(crate) struct CollectorAsyncClientHttp {
     endpoint: Uri,
-    client: HttpClient,
+    #[cfg(feature = "collector_client")]
+    client: isahc::HttpClient,
+    #[cfg(feature = "wasm_collector_client")]
```
Comment (jtescher) [inline opentelemetry-jaeger/src/collector.rs:12]: These features have to be exclusive or else `--all-features` won't compile

Comment (boxdot) [reply opentelemetry-jaeger/src/collector.rs]: Thanks. I now enable the wasm feature only if the bare feature is not set.

**Comment by jtescher** [comment]
@boxdot have you looked more at the linux foundation privacy / CLA situation?

**Comment by boxdot** [comment]
> @boxdot have you looked more at the linux foundation privacy / CLA situation?

Signed. 

Looks like the test coverage got a hit. I did not add any wasm client tests. Technically, the wasm client code is also not compiled in CI since the the features are exclusive and the native client is preferred by default. Tell me if you prefer to have the CI changes in this PR, otherwise I would propose to make them in a consecutive review.

**Review by jtescher — APPROVED** [review]
Looks good, thanks @boxdot

**Comment by jtescher** [comment]
Can do further improvements in subsequent PRs, this looks like a good place to start.

**Pull request merged**